diff --git a/cherokee/cryptor_libssl.c b/cherokee/cryptor_libssl.c
index fabf9912..80a5c2eb 100644
--- a/cherokee/cryptor_libssl.c
+++ b/cherokee/cryptor_libssl.c
@@ -356,7 +356,7 @@ verify_trace_cb(int preverify_ok, X509_STORE_CTX *x509_store)
 		char *ptr;
 		X509_print (mem, peer_certificate);
 		BIO_get_mem_data(mem, &ptr);
-		TRACE (ENTRIES, "SSL: %s", ptr);
+		TRACE (ENTRIES, "SSL: %s\n", ptr);
 		BIO_free (mem);
 	}
 
@@ -785,7 +785,7 @@ _socket_init_tls (cherokee_cryptor_socket_libssl_t *cryp,
 		if (cipher) {
 			SSL_CIPHER_description (cipher, &buf[0], buf_size-1);
 
-			TRACE (ENTRIES, "SSL: %s, %sREUSED, Ciphers: %s",
+			TRACE (ENTRIES, "SSL: %s, %sREUSED, Ciphers: %s\n",
 			       SSL_get_version(cryp->session),
 			       SSL_session_reused(cryp->session)? "" : "Not ", &buf[0]);
 		}
diff --git a/cherokee/post.c b/cherokee/post.c
index 5d98b920..8a1fe4bd 100644
--- a/cherokee/post.c
+++ b/cherokee/post.c
@@ -309,6 +309,7 @@ process_chunk (cherokee_post_t   *post,
 		}
 
 		if (unlikely (p+2 > end)) {
+			cherokee_buffer_clean (in);
 			return ret_ok;
 		}
 
@@ -325,6 +326,7 @@ process_chunk (cherokee_post_t   *post,
 
 		/* Read the length
 		*/
+		errno = 0;
 		content_size = (size_t) strtoul (begin, NULL, 16);
 		if (unlikely (errno != 0)) {
 			return ret_error;
@@ -349,6 +351,7 @@ process_chunk (cherokee_post_t   *post,
 			if (post->chunked.retransmit) {
 				cherokee_buffer_add_str (out, "0" CRLF);
 			}
+			p += 2; /* CRLF */
 			begin = p;
 			break;
 		}
@@ -378,7 +381,7 @@ process_chunk (cherokee_post_t   *post,
 	/* Very unlikely, but still possible
 	 */
 	if (! cherokee_buffer_is_empty(in)) {
-		TRACE (ENTRIES, "There are %d left-over bytes in the post buffer -> incoming header", in->len);
+		TRACE (ENTRIES, "There are %d left-over bytes in the post buffer -> incoming header\n", in->len);
 #if 0
 		cherokee_buffer_add_buffer (&conn->incoming_header, in);
 		cherokee_buffer_clean (in);
diff --git a/cherokee/rule_url_arg.c b/cherokee/rule_url_arg.c
index 4c88f6e1..0cdf8668 100644
--- a/cherokee/rule_url_arg.c
+++ b/cherokee/rule_url_arg.c
@@ -42,6 +42,11 @@ check_argument (cherokee_rule_url_arg_t *rule,
 {
 	int re;
 
+	if (value == NULL) {
+		TRACE (ENTRIES, "Empty parameter value\n");
+		return ret_not_found;
+	}
+
 	/* Check whether it matches
 	 */
 	re = pcre_exec (rule->pcre, NULL,
diff --git a/cherokee/server.c b/cherokee/server.c
index 7afbf19d..64faccdd 100644
--- a/cherokee/server.c
+++ b/cherokee/server.c
@@ -1770,7 +1770,7 @@ cherokee_server_daemonize (cherokee_server_t *srv)
 {
 	pid_t child_pid;
 
-	TRACE (ENTRIES, "server (%p) about to become evil", srv);
+	TRACE (ENTRIES, "server (%p) about to become evil\n", srv);
 
 	child_pid = fork();
 	switch (child_pid) {
